tests/TestCase/Database/ConnectionTest.php

```diff
@@ -1355,4 +1357,16 @@ public function testNoAutomaticReconnect(): void
         $prop->setValue($conn, $oldDriver);
         $conn->rollback();
     }
+
+    public function testRoles(): void
+    {
+        $this->assertSame('write', $this->connection->role());
```

Comment:
Should we use the constant here?

Reply:
Yea ,we can. I was doing a sanity check that I set the names properly everywhere.